functions/src/functions/register-buid.ts

```diff
@@ -0,0 +1,128 @@
+import * as functions from "firebase-functions";
+import * as t from "io-ts"
+import {isLeft} from "fp-ts/lib/Either";
+import * as admin from "firebase-admin";
+import {PathReporter} from "io-ts/lib/PathReporter";
+import {MAX_BUIDS_PER_USER, REGION} from "../settings";
+import {CollectionReference} from "@google-cloud/firestore";
+import {randomBytes} from "crypto";
+
+const MAX_BUID_RETRIES = 10;
+const BUID_BYTE_LENGTH = 10;
+
+function getUTCTimestamp(): number {
```

Comment:
Minor nitpick: there's technically no such thing as a _UTC timestamp_. A correct term here is _Unix timestamp_.